app/models/user.rb

```diff
@@ -0,0 +1,35 @@
+require 'bcrypt'
+
+class User
+
+include DataMapper::Resource
+
+  property :id, Serial
+  property :name, String
+  property :email, String, format: :email_address, required: true, unique: true
+  property :password_digest, BCryptHash
+  property :username, String, unique: true, required: true
+
+  has n, :peeps
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  validates_confirmation_of :password
+
+  def self.authenticate(email, password)
+    user = first(email: email)
+    if user && BCrypt::Password.new(user.password_digest) == password
+      user
+    else
+      nil
+    end
+  end
+
+  class <<self
+    def count
+      self.all.count
+    end
+  end
```

Comment:
This feels like a laborious way of writing this class method. Is there a more idiomatic way?